projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -85,11 +96,42 @@ export class CartesianLegend {
   }
 
   private drawLegendEntries(container: ContainerElement): void {
-    select(container)
+    const containerSelection = select(container);
+    if (!this.isGrouped) {
+      containerSelection
+        .append('div')
+        .classed('legend-entries', true)
+        .selectAll('.legend-entry')
+        .data(this.series.filter(series => !series.hide))
+        .enter()
+        .each((_, index, elements) => this.drawLegendEntry(elements[index]));
+    } else {
+      containerSelection
+        .selectAll('.legend-entries')
+        .data(this.groupedSeries)
+        .enter()
+        .append('div')
+        .attr('class', seriesGroup => `legend-entries group-${seriesGroup[0].assignedGroup!.id}`)
+        .each((seriesGroup, index, elements) => this.drawLegendEntriesTitleAndValues(seriesGroup, elements[index]));
+    }
+  }
+
+  private drawLegendEntriesTitleAndValues(seriesGroup: Series<{}>[], element: HTMLDivElement): void {
+    const legendEntriesSelection = select(element);
+    legendEntriesSelection
+      .selectAll('.legend-entries-title')
+      .data([seriesGroup])
+      .enter()
+      .append('div')
+      .classed('legend-entries-title', true)
+      .text(group => `${group[0].name}:`)
+      .on('click', () => this.updateActiveSeries(seriesGroup));
+
+    legendEntriesSelection
```

Comment:
This append is not bound by any data, so it will always be applied. If the legends dom are not cleared then this would add duplicate nodes.

Also, why are appending `div` inside an svg?

Reply:
hmm, looks like all of the legend is in div and spans. Please confirm.

Reply:
It has `svg` but not as parent. `Visualizations` are `svgs` then `legend` is `div`. inside that we have `divs` and `spans`, and `svgs` (used for `dot` part).